### PR TITLE
installation-device.nix: explain sshd usage, don't include clone-config

### LINKED ISSUE #26776: SSHD doesn't work.
### Steps to reproduce
1:
`cat ./custom.nix`
```
{ config, lib, pkgs, ... }:

{
  imports = [ ./modules/installer/cd-dvd/installation-cd-base.nix ];

        users.extraUsers.root.openssh.authorizedKeys.keys = [ "ssh-ed25519 AAAA... sorcus@laptop" ];

        systemd.network.enable = true;
        systemd.network.networks.wired.name = "enp0s3";
        systemd.network.networks.wired.address = [ "2a01:***:***::13/64" ];
        systemd.network.networks.wired.gateway = [ "fe80::1" ];

        services.openssh.enable = true;
}
```
2:
`nix-build -A config.system.build.isoImage -I nixos-config=./custom.nix ./default.nix`

3:
Create virtual machine and run with nixos iso. SSH Daemon doesn't work. It's loaded, but not active.

4:
But it's working with `services.openssh.startWhenNeeded = true;` ... and show this warnings:
```
Jun 22 21:19:09 nixos systemd[1]: sshd.service: Service lacks both ExecStart= and ExecStop= setting. Refusing.
Jun 22 21:07:44 nixos systemd[1]: Started Session 3 of user root.
Jun 22 21:07:50 nixos systemd[1]: sshd.service: Service lacks both ExecStart= and ExecStop= setting. Refusing.
```

## Technical details

* System: 17.09.git.c89efa3 (Hummingbird)
* Nix version: nix-env (Nix) 1.11.11
* Nixpkgs version: 17.09.git.c89efa3
* Sandboxing enabled: build-use-sandbox = false

### LINKED ISSUE #63576: sd-image-aarch64 shouldn't import profiles/installation-device.nix
## Issue description
When downloading the aarch64 sdcard image, the `/etc/nixos/configuration.nix` provided there imports `nixos/modules/installer/cd-dvd/sd-image-aarch64.nix`, which imports `nixos/modules/profiles/installation-device.nix`, which (among others) enables the nixos manual on tty8, force-enables the documentation, disabes sudo and udisks, disables sshd on bootup and enables automatic login as root without password.

We probably want to provide a different configuration.nix in `/etc/nixos/configuration.nix` not importing installer-related things (or at least make it easy to disable them). 

### Steps to reproduce
 - Download the aarch64 sdcard image
 - flash it to a raspberry pi3
 - edit /etc/nixos/configuration.nix (add `services.openssh.enable = true` for example)
 - call `nixos-rebuild switch`
 - wonder why it's not started

## Technical details
`nixos-sd-image-19.09pre183392.83ba5afcc96-aarch64-linux.img`

cc @NinjaTrappeur

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
The configuration used to build an installation medium isn't really
suitable to be (re)used when building a real system.
    
For example, it enables automatic root login without a password on the
console, installs sshd, but prevents it from starting on boot.
    
This can be very misleading when users try to derive their real system
from it - they should use nixos-generate-config instead, as described in
the manual.

Also update the mingetty helpLine to describe the sshd setup.
  
Closes #63576
Closes #26776

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by flokli** [comment]
@GrahamcOfBorg test installer

**Comment by samueldr** [comment]
@GrahamcOfBorg eval

??!?

File: nixos/modules/profiles/installation-device.nix [hunk 1]
```diff
@@ -10,10 +10,6 @@ with lib;
       ../installer/scan/detected.nix
       ../installer/scan/not-detected.nix
 
-      # Allow "nixos-rebuild" to work properly by providing
-      # /etc/nixos/configuration.nix.
-      ./clone-config.nix
-
```
Comment (samueldr) [inline nixos/modules/profiles/installation-device.nix]: `nixos/modules/installer/virtualbox-demo.nix` was using `installer.cloneConfigExtra` from this import.

```
~/tmp/nixpkgs/nixpkgs-PR63773 2 $ nix-instantiate --arg nixpkgs '{ outPath=./.; revCount=999999; shortRev="ofborg"; }' .
/nixos/release-combined.nix -A tested
warning: dumping very large path (> 256 MiB); this may run out of memory
error: The option `installer' defined in `/Users/samuel/tmp/nixpkgs/nixpkgs-PR63773/nixos/modules/installer/virtualbox-demo.nix'
does not exist.
(use '--show-trace' to show detailed location information)
```

In addition, the cloneConfig system for installer isos is desired. Let's say I need to rebuild with a specific setting for doing something, there is currently a loss of function.

I think, instead, `installer.cloneConfig` should be set to `false` on image types where the installation media is also the installation target.

**Review by samueldr — CHANGES_REQUESTED** [review]

File: nixos/modules/profiles/installation-device.nix (hunk 1 above)
Comment (flokli) [reply nixos/modules/profiles/installation-device.nix]: I pushed a new version, which sets `installer.cloneConfig = false;` instead.

This only seemed to be necessary for `sd-image.nix`.

**Comment by worldofpeace** [comment]
Oh, missed the `profiles` documentation because I don't think I noticed when this was added.

File: nixos/modules/profiles/installation-device.nix [hunk 2]
```diff
@@ -55,13 +55,13 @@ with lib;
...
-      ''
-
-        The "nixos" and "root" account have empty passwords.  ${
-          optionalString config.services.xserver.enable
-            "Type `sudo systemctl start display-manager' to\nstart the graphical user interface."}
-      '';
+    services.mingetty.helpLine = ''
+      The "nixos" and "root" accounts have empty passwords.
+      sshd can be started manually via 'systemctl start sshd'
+      - remember to set a password, or add to ~/.ssh/authorized_keys first.
```
Comment (worldofpeace) [inline nixos/modules/profiles/installation-device.nix]: How about
```
The "nixos" and "root" accounts have empty passwords.

Type `sudo systemctl start sshd` to the SSH daemon.
You then must set a password for either "root"  or "nixos" with `passwd` to be able to login.
```

That's inline to what's in the manual, but we don't have an entry there about `authorized_keys`.
Also it may become confusing as to which `authorized_keys` file to add to since it could either be `root`'s home `nixos`'s.

File: nixos/modules/profiles/installation-device.nix [hunk 3]
```diff
@@ -55,13 +55,13 @@ with lib;
...
+    '' + optionalString config.services.xserver.enable ''
+          Type `systemctl start display-manager' to\nstart the graphical user interface.
```
Comment (worldofpeace) [inline nixos/modules/profiles/installation-device.nix]:
```suggestion
          Type `sudo systemctl start display-manager' to\nstart the graphical user interface.
```

File: nixos/doc/manual/configuration/profiles/installation-device.xml [hunk 4]
```diff
@@ -6,33 +6,31 @@
...
+  <literal>sudo</literal> is possible.
+  <link linkend="opt-networking.wireless.enable">wpa_supplicant</link> is
+  enabled, but configured to not autostart.
  </para>
-
  <para>
-  A message is shown to the user to start a display manager if needed, ssh with
-  <xref linkend="opt-services.openssh.permitRootLogin"/> are enabled (but
-  doesn't autostart). WPA Supplicant is also enabled without autostart.
+  A message is shown to the user that he can log in either as
```
Comment (worldofpeace) [inline nixos/doc/manual/configuration/profiles/installation-device.xml]:
```suggestion
  A message is shown to the user that they can log in either as
```

File: nixos/doc/manual/configuration/profiles/installation-device.xml [hunk 5]
```diff
@@ -6,33 +6,31 @@
  <title>Installation Device</title>
 
  <para>
-  Provides a basic configuration for installation devices like CDs. This means
-  enabling hardware scans, using the <link linkend="sec-profile-clone-config">
-  Clone Config profile</link> to guarantee
-  <filename>/etc/nixos/configuration.nix</filename> exists (for
-  <command>nixos-rebuild</command> to work), a copy of the Nixpkgs channel
-  snapshot used to create the install media.
+  Provides a basic configuration for installation devices like CDs.
```
Comment (worldofpeace) [inline nixos/doc/manual/configuration/profiles/installation-device.xml]: At this point `installation device` as a CD even feels antiquated :smile: .
Though I think renaming that is out of scope.

File: nixos/doc/manual/configuration/profiles/installation-device.xml (hunk 4 above)
Comment (flokli) [reply nixos/doc/manual/configuration/profiles/installation-device.xml]: I went with

> A message is shown to the user to log in either as […]

File: nixos/doc/manual/configuration/profiles/installation-device.xml (hunk 5 above)
Comment (flokli) [reply nixos/doc/manual/configuration/profiles/installation-device.xml]: Haha yes, the code is also in the `nixos/modules/installer/cd-dvd/` subfolder (and contains `sd-image` and `system-tarball` too) - But I guess cleaning this up should be out of scope for this PR at least.

File: nixos/doc/manual/configuration/profiles/installation-device.xml [hunk 6]
```diff
@@ -6,33 +6,31 @@
...
+  This enables redistributable firmware, includes the
+  <link linkend="sec-profile-clone-config">Clone Config profile</link>
+  and a copy of the Nixpkgs channel, so nixos-install works out of the box.
```
Comment (worldofpeace) [inline nixos/doc/manual/configuration/profiles/installation-device.xml]:
```suggestion
  and a copy of the Nixpkgs channel, so <command>nixos-install</command> works out of the box.
```

File: nixos/doc/manual/configuration/profiles/installation-device.xml [hunk 7]
```diff
@@ -6,33 +6,31 @@
...
  </para>
-
  <para>
-  Additionally, documentation for <link linkend="opt-documentation.enable">
-  Nixpkgs</link> and <link linkend="opt-documentation.nixos.enable">NixOS
-  </link> are forcefully enabled (to override the
+  Documentation for <link linkend="opt-documentation.enable"> Nixpkgs</link>
```
Comment (worldofpeace) [inline nixos/doc/manual/configuration/profiles/installation-device.xml]:
```suggestion
  Documentation for <link linkend="opt-documentation.enable">Nixpkgs</link>
```
space needed?

File: nixos/doc/manual/configuration/profiles/installation-device.xml [hunk 8]
```diff
@@ -6,33 +6,31 @@
...
+  and <link linkend="opt-documentation.nixos.enable">NixOS</link> are
+  forcefully enabled (to override the
   <link linkend="sec-profile-minimal">Minimal profile</link> preference); the
-  NixOS manual is shown automatically on TTY 8, sudo and udisks are disabled.
-  Autologin is enabled as root.
+  NixOS manual is shown automatically on TTY 8, udisks is disabled.
+  Autologin is enabled as <literal>nixos</literal> user, while passwordless
+  <literal>sudo</literal> is possible.
```
Comment (worldofpeace) [inline nixos/doc/manual/configuration/profiles/installation-device.xml]: Is it mentioned anywhere that "root" hasn't a password?

Comment (flokli) [reply nixos/doc/manual/configuration/profiles/installation-device.xml]: yes, but a tad late. I reshuffled things a bit, might look better now.

File: nixos/doc/manual/configuration/profiles/installation-device.xml [hunk 9]
```diff
@@ -6,33 +6,31 @@
...
+  It is explained how to login, start the ssh server, and if available,
+  how to start the display manager.
  </para>
 
  <para>
-  Finally, vim is installed, root is set to not have a password, the kernel is
-  made more silent for remote public IP installs, and several settings are
-  tweaked so that the installer has a better chance of succeeding under
-  low-memory environments.
+  Several settings are tweaked so that the installer has a better chance of
```
Comment (worldofpeace) [inline nixos/doc/manual/configuration/profiles/installation-device.xml]: I'm not sure if we should imply for success but to just improve it's performance in that environment.

```
Several settings are tweaked to improve Nix and the installers performance under memory-constrained environments.
```

File: nixos/modules/profiles/installation-device.nix [hunk 10]
```diff
@@ -55,13 +55,15 @@ with lib;
...
-      ''
-
-        The "nixos" and "root" account have empty passwords.  ${
-          optionalString config.services.xserver.enable
-            "Type `sudo systemctl start display-manager' to\nstart the graphical user interface."}
-      '';
+    services.mingetty.helpLine = ''
+      The "nixos" and "root" accounts have empty passwords.
+
+      Type `sudo systemctl start sshd` to the SSH daemon.
```
Comment (worldofpeace) [inline nixos/modules/profiles/installation-device.nix]:
```suggestion
      Type `sudo systemctl start sshd` to start the SSH daemon.
```

File: nixos/modules/profiles/installation-device.nix [hunk 11]
```diff
@@ -55,13 +55,15 @@ with lib;
...
+      You then must set a password for either "root" or "nixos"
+      with `passwd` to be able to login.
+    '' + optionalString config.services.xserver.enable ''
+          Type `sudo systemctl start display-manager' to\nstart the graphical user interface.
```
Comment (worldofpeace) [inline nixos/modules/profiles/installation-device.nix]: The indentation here looks a bit larger than standard.

File: nixos/doc/manual/configuration/profiles/installation-device.xml (hunk 9 above)
Comment (flokli) [reply nixos/doc/manual/configuration/profiles/installation-device.xml]: This could be misinterpreted at "making it faster", but it seems more to be a "getting it to work at all".

File: nixos/modules/profiles/installation-device.nix (hunk 11 above)
Comment (flokli) [reply nixos/modules/profiles/installation-device.nix]: fixed, thanks!

File: nixos/modules/profiles/installation-device.nix (hunk 10 above)
Comment (flokli) [reply nixos/modules/profiles/installation-device.nix]: fixed, thanks!

**Review by worldofpeace — APPROVED** [review]
Docs seem good :+1:

**Comment by flokli** [comment]
@samueldr, could you give this a try once `sd_image` builds again?

**Comment by worldofpeace** [comment]
Probably immediately easy enough to revert https://github.com/NixOS/nixpkgs/pull/63835 for the image

**Comment by flokli** [comment]
Rebased on top of master, as https://github.com/NixOS/nixpkgs/pull/66850 landed.
